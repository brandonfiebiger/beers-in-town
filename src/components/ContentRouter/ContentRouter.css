header button {
  height: 3.3rem;
  width: 10.5rem;
  margin-top: 1rem;
  margin-right: .2rem;
  border-radius: 10px;
  text-decoration: underline;
  font-size: 1.3rem;
  background: rgba(25, 25, 25, .7);
  box-shadow: 0 0 80px #000000;
  cursor: pointer;
  color: #f9f9f9
  
}

header button:hover {
  background: rgba(39, 14, 14, 0.9);
}

.selected {
  background: rgba(39, 14, 14, 0.9);
}


header {
  display: flex;
  margin: 0 auto;
  width: 35rem;
  justify-content: space-around;
}

input {
  height: 1.4rem;
  margin-right: .2rem;
}

.routes-container {
  overflow: scroll;
}

@media screen and (max-width: 560px) {
  header {
    flex-direction: row;
    width: 100vw;
    text-align: center;
    align-items: center;
  }

  header button {
    margin-right: 0;
    width: 7rem;
  }

  .location-inputs {
    width: 18rem;
  }

  .search-button {
    width: 18rem;
  }
}