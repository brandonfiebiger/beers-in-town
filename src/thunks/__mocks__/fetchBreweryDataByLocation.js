export const fetchBreweryDataByLocation = jest.fn()