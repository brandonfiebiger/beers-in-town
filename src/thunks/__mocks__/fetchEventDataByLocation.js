export const fetchEventDataByLocation = jest.fn()