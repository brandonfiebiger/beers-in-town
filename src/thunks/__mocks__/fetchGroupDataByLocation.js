export const fetchGroupDataByLocation = jest.fn()