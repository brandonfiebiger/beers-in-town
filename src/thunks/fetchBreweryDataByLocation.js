import { apikey } from '../utils/variables'
import { populateBreweriesFromLocation } from '../actions/';

export const fetchBreweryDataByLocation = (latitude, longitude) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/find/venues?key=${apikey}&sign=true&photo-host=secure&text=brewery&lon=${longitude}&radius=100&lat=${latitude}&page=200`);
      //add condition to check that response is ok check thunk lesson
      const breweries = await response.json();
      dispatch(populateBreweriesFromLocation(breweries))
    } catch (error) {
       alert('could not get your location')
    }
  }
}
