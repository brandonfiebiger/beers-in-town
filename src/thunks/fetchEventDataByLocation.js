import { populateEventsFromLocation } from '../actions/';
import { apikey } from '../utils/variables';
import { eventCleaner } from '../utils/helper';

export const fetchEventDataByLocation = (latitude, longitude) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/find/upcoming_events?key=${apikey}&sign=true&photo-host=public&lon=${longitude}&page=2000&text=craft beer, brewery&radius=100&lat=${latitude}`);
      const events = await response.json();
      const cleanEvents = await eventCleaner(events.events);
      dispatch(populateEventsFromLocation(cleanEvents))
    } catch (error) {
       alert('Could not get your location')
    }
  }
}


