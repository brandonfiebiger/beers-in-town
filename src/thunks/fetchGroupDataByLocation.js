import { apikey } from '../utils/variables';
import  { populateGroupsFromLocation } from '../actions/';
import { cleanGroupData } from '../utils/helper';


export const fetchGroupDataByLocation = (latitude, longitude) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/find/groups?key=${apikey}&sign=true&photo-host=public&lon=${longitude}&text=craft beer&radius=100&lat=${latitude}&page=100`);
      const groups = await response.json();
      const cleanedGroups = await cleanGroupData(groups);
      dispatch(populateGroupsFromLocation(cleanedGroups))
    } catch (error) {
      throw new Error(error.message)
    }
  }
}

