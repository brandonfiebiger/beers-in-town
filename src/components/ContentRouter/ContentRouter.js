import React, { Component } from 'react';
import { Route, withRouter, NavLink } from 'react-router-dom';
import  EventContainer  from '../../containers/EventContainer/EventContainer';
import BreweryContainer  from '../../containers/BreweryContainer/BreweryContainer';
import GroupsContainer from '../../containers/GroupsContainer/GroupsContainer';
import EventInfo from '../../containers/EventInfo/EventInfo';
import GroupInfo from '../../containers/GroupInfo/GroupInfo';
import { connect } from 'react-redux';
import { fetchBreweryDataByLocation } from '../../thunks/fetchBreweryDataByLocation';
import { fetchEventDataByLocation } from '../../thunks/fetchEventDataByLocation';
import { fetchGroupDataByLocation } from '../../thunks/fetchGroupDataByLocation';
import './ContentRouter.css';

export class ContentRouter extends Component {
  constructor() {
    super()

    this.state = {
      selected: ''
    }
  }

 handleBreweryRoute = () => {
    const { location, populateBreweries, history, breweries } = this.props;
    if (!breweries.length) {
      populateBreweries(location.latitude, location.longitude);
    }
    history.push('/breweries');
    this.setState({
      selected: 'breweries'
    })
  }

 handleEventRoute = () => {
    const { location, populateEvents, history, events } = this.props;
    if (!events.length) {
      populateEvents(location.latitude, location.longitude);
    }
    history.push('/events');
    this.setState({
      selected: 'events'
    })
  }

 handleGroupsRoute = () => {
    const { location, populateGroups, history, groups } = this.props;
    if (!groups.length) {
      populateGroups(location.latitude, location.longitude);
    }
    history.push('/groups')
    this.setState({
      selected: 'groups'
    })
  }

  render() {
    return(
      <div>
      <header>
        <button className={this.state.selected === 'events' ? "event-button selected" : "event-button"} onClick={() => this.handleEventRoute()}>events</button>
        <button className={this.state.selected === 'breweries' ? "brewery-button selected" : "brewery-button"} onClick={() => this.handleBreweryRoute()}>breweries</button>
        <button className={this.state.selected === 'groups' ? "group-button selected" : "group-button"} onClick={() => this.handleGroupsRoute()}>groups</button>
      </header>
        <Route exact path= '/events' component={EventContainer} />
        <Route exact path= '/breweries' component={BreweryContainer} />
        <Route exact path= '/groups' component={GroupsContainer} />
        <Route exact path={`/events/${this.props.eventProps.id}`} render={() => <EventInfo {...this.props} />}/>
        <Route exact path={`/groups/${this.props.groupProps.id}`} render={() => <GroupInfo {...this.props} />}/>
      </div>
    )
  }

}



export const mapStateToProps = state => ({
  eventProps: state.eventToView,
  groupProps: state.groupToView,
  location: state.location,
  events: state.events,
  groups: state.groups,
  breweries: state.breweries
})

export const mapDispatchToProps = dispatch => ({
  populateEvents: (latitude, longitude) => dispatch(fetchEventDataByLocation(latitude, longitude)),
  populateGroups: (latitude, longitude) => dispatch(fetchGroupDataByLocation(latitude, longitude)),
  populateBreweries: (latitude, longitude) => dispatch(fetchBreweryDataByLocation(latitude, longitude))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ContentRouter));