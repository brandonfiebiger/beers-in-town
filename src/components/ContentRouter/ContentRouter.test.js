import { ContentRouter, mapDispatchToProps, mapStateToProps } from './ContentRouter';
import ReactDOM from 'react-dom';
import React from 'react';
import { shallow, mount } from 'enzyme';
import createRouterContext from 'react-router-test-context';
import { fetchBreweryDataByLocation } from '../../thunks/fetchBreweryDataByLocation';
import { fetchEventDataByLocation } from '../../thunks/fetchEventDataByLocation';
import { fetchGroupDataByLocation } from '../../thunks/fetchGroupDataByLocation';
jest.mock('../../thunks/fetchGroupDataByLocation')
jest.mock('../../thunks/fetchEventDataByLocation');
jest.mock('../../thunks/fetchBreweryDataByLocation');

describe('ContentRouter', () => {

  let wrapper;
  let mockEventProps;
  let mockGroupProps;
  let mockLocation;
  let mockHistory;
  let mockPopulateBreweries;

  beforeEach(() => {
    mockEventProps = {
      id: 5
    }
    
    mockHistory = { push: jest.fn() };

    mockLocation = {
      latitude: 5,
      longitude: 10
    }
    
    mockGroupProps = {
      id: 80
    }

    mockPopulateBreweries = jest.fn()

    wrapper = shallow(<ContentRouter events={[]} eventProps={ mockEventProps } groupProps={ mockGroupProps } breweries={ [] } location={ mockLocation } populateEvents={ mockPopulateBreweries } populateBreweries={ mockPopulateBreweries } history={ mockHistory }/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handleEventRoute when event-button is clicked', () => {
    wrapper.instance().handleEventRoute = jest.fn()
    wrapper.find('.event-button').simulate('click');
    expect(wrapper.instance().handleEventRoute).toHaveBeenCalled();
  })

  it('should call handleGroupsRoute when event-button is clicked', () => {
    wrapper.instance().handleGroupsRoute = jest.fn()
    wrapper.find('.group-button').simulate('click');
    expect(wrapper.instance().handleGroupsRoute).toHaveBeenCalled();
  })

  it('should call handleBreweryRoute when event-button is clicked', () => {
    wrapper.instance().handleBreweryRoute = jest.fn()
    wrapper.find('.brewery-button').simulate('click');
    expect(wrapper.instance().handleBreweryRoute).toHaveBeenCalled();
  })

  describe('handleBreweryRoute', () => {
   it('should dispatch populateBreweries if there are no breweries in array', () => {
    const context = createRouterContext();
     const mockPopulateBreweries = jest.fn()
     wrapper = shallow(<ContentRouter eventProps={ mockEventProps } groupProps={ mockGroupProps } breweries={ [] } location={ mockLocation } populateBreweries={ mockPopulateBreweries } populateEvents={ jest.fn() } populateGroups={ jest.fn() } history={ mockHistory } events={ [] }/>, { context })
     wrapper.instance().handleBreweryRoute()
    
     expect(mockPopulateBreweries).toHaveBeenCalled()
    // wrapper.instance().handleEventRoute()
    // expect(wrapper.props().populateEvents).toHaveBeenCalled()
   })
  })

  describe('handleEventRoute', () => {
    it('should dispatch populateBreweries if there are no breweries in array', () => {
     const context = createRouterContext();
      const mockPopulateEvents = jest.fn()
      wrapper = shallow(<ContentRouter eventProps={ mockEventProps } groupProps={ mockGroupProps } breweries={ [] } location={ mockLocation } populateBreweries={ mockPopulateBreweries } populateEvents={ mockPopulateEvents } populateGroups={ jest.fn() } history={ mockHistory } events={ [] }/>, { context })
     wrapper.instance().handleEventRoute()
     expect(mockPopulateEvents).toHaveBeenCalled()
    })
   })

   describe('handleGroupRoute', () => {
    it('should dispatch populateGroups if there are no breweries in array', () => {
     const context = createRouterContext();
      const mockPopulateGroups = jest.fn()
      wrapper = shallow(<ContentRouter eventProps={ mockEventProps } groupProps={ mockGroupProps } breweries={ [] } location={ mockLocation } populateBreweries={ mockPopulateBreweries } populateEvents={ jest.fn() } populateGroups={ mockPopulateGroups } history={ mockHistory } events={ [] } groups={ [] }/>, { context })
     wrapper.instance().handleGroupsRoute()
     expect(mockPopulateGroups).toHaveBeenCalled()
    })
   })


  describe('mapDispatchToProps', () => {
    it('calls dispatch with a fetchBreweryDataByLocation action', () => {
      const mockDispatch = jest.fn();

      const actionToDispatch = fetchBreweryDataByLocation(5, 10)
      
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.populateBreweries('Arlen', 'Texas');

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with a fetchEventDataByLocation action', () => {
      const mockDispatch = jest.fn();

      const actionToDispatch = fetchBreweryDataByLocation(5, 10)
      
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.populateEvents(2, 5);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with a fetchEventDataByLocation action', () => {
      const mockDispatch = jest.fn();

      const actionToDispatch = fetchGroupDataByLocation(5, 10)
      
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.populateGroups(2, 5);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with eventProps, groupProps, location, events, groups and breweries', () => {
      const mockState = {
        eventProps: {
          id: 3
        },
        groupProps: {
          id:2
        },
        location: {latitude: 3, longitude: 3},
        events: [],
        groups: [],
        breweries: []
      }
      const expected = {
        eventProps: undefined,
        groupProps: undefined,
        location: {latitude: 3, longitude: 3},
        events: [],
        groups: [],
        breweries: []
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected);
    })
  })
})