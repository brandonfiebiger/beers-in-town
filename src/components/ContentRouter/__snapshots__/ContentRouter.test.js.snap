// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentRouter should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentRouter
    breweries={Array []}
    eventProps={
        Object {
            "id": 5,
          }
    }
    events={Array []}
    groupProps={
        Object {
            "id": 80,
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
    location={
        Object {
            "latitude": 5,
            "longitude": 10,
          }
    }
    populateBreweries={[Function]}
    populateEvents={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <button
                    className="event-button"
                    onClick={[Function]}
          >
                    events
          </button>
          <button
                    className="brewery-button"
                    onClick={[Function]}
          >
                    breweries
          </button>
          <button
                    className="group-button"
                    onClick={[Function]}
          >
                    groups
          </button>
</header>,
        <Route
          component={[Function]}
          exact={true}
          path="/events"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/breweries"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/groups"
/>,
        <Route
          exact={true}
          path="/events/5"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/groups/80"
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="event-button"
              onClick={[Function]}
>
              events
</button>,
            <button
              className="brewery-button"
              onClick={[Function]}
>
              breweries
</button>,
            <button
              className="group-button"
              onClick={[Function]}
>
              groups
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "events",
              "className": "event-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "events",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "breweries",
              "className": "brewery-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "breweries",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "groups",
              "className": "group-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "groups",
            "type": "button",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/events",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/breweries",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/groups",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/events/5",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/groups/80",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <button
                        className="event-button"
                        onClick={[Function]}
            >
                        events
            </button>
            <button
                        className="brewery-button"
                        onClick={[Function]}
            >
                        breweries
            </button>
            <button
                        className="group-button"
                        onClick={[Function]}
            >
                        groups
            </button>
</header>,
          <Route
            component={[Function]}
            exact={true}
            path="/events"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/breweries"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/groups"
/>,
          <Route
            exact={true}
            path="/events/5"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/groups/80"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="event-button"
                onClick={[Function]}
>
                events
</button>,
              <button
                className="brewery-button"
                onClick={[Function]}
>
                breweries
</button>,
              <button
                className="group-button"
                onClick={[Function]}
>
                groups
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "events",
                "className": "event-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "events",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "breweries",
                "className": "brewery-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "breweries",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "groups",
                "className": "group-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "groups",
              "type": "button",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/events",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/breweries",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/groups",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/events/5",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/groups/80",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
